language-plutus-core/src/Language/PlutusCore/Constant/Dynamic/Instances.hs

```diff
@@ -53,13 +53,7 @@ instance KnownType a => KnownType (EvaluationResult a) where
     makeKnown EvaluationFailure     = Error () $ toTypeAst @a Proxy
     makeKnown (EvaluationSuccess x) = makeKnown x
 
-    -- There are two 'EvaluationResult's here: an external one (which any 'KnownType'
-    -- instance has to deal with) and an internal one (specific to this particular instance).
-    -- Our approach is to always return 'EvaluationSuccess' for the external 'EvaluationResult'
-    -- and catch all 'EvaluationFailure's in the internal 'EvaluationResult'.
-    -- This allows *not* to short-circuit when 'readKnown' fails to read a Haskell value.
-    -- Instead the user gets an explicit @EvaluationResult a@ and evaluation proceeds normally.
-    readKnown eval = mapDeepReflectT (fmap $ EvaluationSuccess . sequence) . readKnown eval
+    readKnown _ _ = throwingWithCause _UnliftingError "Catching errors is not supported" Nothing
```

Comment:
This deletes support for exception handling in PLC. It's harder to support it with the new machinery and since we don't support exception handling elsewhere, I guess no point to complicate the code.